---
title: "bcl11a_analysis_and_viz"
author: "Dustin Tillman"
date: '2022-07-18'
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
library(ggpubr)
library(Mus.musculus)
library(Homo.sapiens)
library(biomaRt)
library(SummarizedExperiment)
library(DESeq2)
library(apeglm)
library(pheatmap)
library(grid)
library(RColorBrewer)
library(ggrepel)
library(clusterProfiler)
library(VennDiagram)
library(tidyverse)
cbPalette <- c("#FFFFFF", # white 
               "#E69F00", # orange
               "#56B4E9", # light blue
               "#009E73", # green
               "#CC79A7", # pink
               "#D55E00", # red
               "#0072B2", # dark blue
               "#F0E442") # yellow
boldBlack_theme <- function(base_size = 12, base_family = "") {
  ggplot2::theme_grey(
    base_size = base_size,
    base_family = base_family
  ) %+replace%
    ggplot2::theme(
      # Specify axis options
      axis.line = ggplot2::element_blank(),
      #axis.text.x = ggplot2::element_blank(),
      axis.text.x = ggplot2::element_text(
        size = base_size * 0.8,
        face = "bold",
        color = "white",
        lineheight = 0.9
      ),
      axis.text.y = ggplot2::element_text(
        size = base_size * 0.8,
        face = "bold",
        color = "white",
        lineheight = 0.9
      ),
      axis.ticks = ggplot2::element_line(
        color = "white",
        size = 0.2
      ),
      axis.title.x = ggplot2::element_text(
        size = base_size,
        color = "white",
        face = "bold",
        margin = margin(
          0, 10, 0,
          0
        )
      ),
      axis.title.y = ggplot2::element_text(
        size = base_size,
        face = "bold",
        color = "white", angle = 90,
        margin = margin(
          0, 10, 0,
          0
        )
      ),
      axis.ticks.length = ggplot2::unit(0.3, "lines"),
      # Specify legend options
      legend.background = ggplot2::element_rect(
        color = NA,
        fill = "black"
      ),
      legend.key = ggplot2::element_rect(
        color = "white",
        fill = "black"
      ),
      legend.key.size = ggplot2::unit(1.2, "lines"),
      legend.key.height = NULL,
      legend.key.width = NULL,
      legend.text = ggplot2::element_text(
        size = base_size * 0.8,
        color = "white",
        face = "bold"
      ),
      legend.title = ggplot2::element_text(
        size = base_size * 0.8,
        face = "bold", hjust = 0,
        color = "white"
      ),
      legend.position = "right",
      legend.text.align = NULL,
      legend.title.align = NULL,
      legend.direction = "vertical",
      legend.box = NULL,
      
      # Specify panel options
      panel.background = ggplot2::element_rect(
        fill = "black",
        color = NA
      ),
      panel.border = ggplot2::element_rect(fill = NA, color = "white"),
      panel.grid.major = ggplot2::element_blank(),
      panel.grid.minor = ggplot2::element_blank(),
      panel.spacing = ggplot2::unit(0.5, "lines"),
      
      # Specify faceting options
      strip.background = ggplot2::element_rect(
        fill = "grey30",
        color = "grey10"
      ),
      strip.text.x = ggplot2::element_text(
        size = base_size * 0.8,
        face = "bold",
        color = "white"
      ),
      strip.text.y = ggplot2::element_text(
        size = base_size * 0.8,
        color = "white",
        face = "bold",
        angle = -90
      ),
      # Specify plot options
      plot.background = ggplot2::element_rect(
        color = "black",
        fill = "black"
      ),
      plot.title = ggplot2::element_text(
        size = base_size * 1.2,
        face = "bold",
        color = "white"
      ),
      plot.margin = ggplot2::unit(rep(1, 4), "lines")
    )
}
```

## Loading in HaplotypeCaller Output

```{r hcLoad, include = F}
sample1 <- read.table("tables_haplotypes/sample1_filtered.haplotypes.table", header = T) %>%
  separate("sample1.AD", c("RC", "AC_1", "AC_2"), 
           sep = ",", convert = T, fill = "right") %>%
  mutate(AC_2_0 = replace(AC_2, is.na(AC_2), 0), .keep = "unused") %>%
  mutate(AC = AC_1 + AC_2_0, .keep = "unused") %>%
  mutate(TC = RC + AC, percentAC = AC/TC) %>%
  mutate(compartment = "soma", genotype = "wt", replicate = 1, sample = 1,
         batch = 1, CD1_supp = F, percentCD1Mass = 0) %>%
  unite("location", CHROM:POS, sep = ".", remove = F)

sample2 <- read.table("tables_haplotypes/sample2_filtered.haplotypes.table", header = T) %>%
  separate("sample2.AD", c("RC", "AC_1", "AC_2"), 
           sep = ",", convert = T, fill = "right") %>%
  mutate(AC_2_0 = replace(AC_2, is.na(AC_2), 0), .keep = "unused") %>%
  mutate(AC = AC_1 + AC_2_0, .keep = "unused") %>%
  mutate(TC = RC + AC, percentAC = AC/TC) %>%
  mutate(compartment = "soma", genotype = "wt", replicate = 2, sample = 2,
         batch = 2, CD1_supp = F, percentCD1Mass = 0) %>%
  unite("location", CHROM:POS, sep = ".", remove = F)

sample3 <- read.table("tables_haplotypes/sample3_filtered.haplotypes.table", header = T) %>%
  separate("sample3.AD", c("RC", "AC_1", "AC_2"), 
           sep = ",", convert = T, fill = "right") %>%
  mutate(AC_2_0 = replace(AC_2, is.na(AC_2), 0), .keep = "unused") %>%
  mutate(AC = AC_1 + AC_2_0, .keep = "unused") %>%
  mutate(TC = RC + AC, percentAC = AC/TC) %>%
  mutate(compartment = "soma", genotype = "wt", replicate = 3, sample = 3,
         batch = 3, CD1_supp = F, percentCD1Mass = 0) %>%
  unite("location", CHROM:POS, sep = ".", remove = F)

sample4 <- read.table("tables_haplotypes/sample4_filtered.haplotypes.table", header = T) %>%
  separate("sample4.AD", c("RC", "AC_1", "AC_2"), 
           sep = ",", convert = T, fill = "right") %>%
  mutate(AC_2_0 = replace(AC_2, is.na(AC_2), 0), .keep = "unused") %>%
  mutate(AC = AC_1 + AC_2_0, .keep = "unused") %>%
  mutate(TC = RC + AC, percentAC = AC/TC) %>%
  mutate(compartment = "soma", genotype = "het", replicate = 1, sample = 4,
         batch = 4, CD1_supp = F, percentCD1Mass = 0) %>%
  unite("location", CHROM:POS, sep = ".", remove = F)

sample5 <- read.table("tables_haplotypes/sample5_filtered.haplotypes.table", header = T) %>%
  separate("sample5.AD", c("RC", "AC_1", "AC_2"), 
           sep = ",", convert = T, fill = "right") %>%
  mutate(AC_2_0 = replace(AC_2, is.na(AC_2), 0), .keep = "unused") %>%
  mutate(AC = AC_1 + AC_2_0, .keep = "unused") %>%
  mutate(TC = RC + AC, percentAC = AC/TC) %>%
  mutate(compartment = "soma", genotype = "het", replicate = 2, sample = 5,
         batch = 5, CD1_supp = F, percentCD1Mass = 0) %>%
  unite("location", CHROM:POS, sep = ".", remove = F)

sample6 <- read.table("tables_haplotypes/sample6_filtered.haplotypes.table", header = T) %>%
  separate("sample6.AD", c("RC", "AC_1", "AC_2"), 
           sep = ",", convert = T, fill = "right") %>%
  mutate(AC_2_0 = replace(AC_2, is.na(AC_2), 0), .keep = "unused") %>%
  mutate(AC = AC_1 + AC_2_0, .keep = "unused") %>%
  mutate(TC = RC + AC, percentAC = AC/TC) %>%
  mutate(compartment = "soma", genotype = "het", replicate = 3, sample = 6,
         batch = 6, CD1_supp = F, percentCD1Mass = 0) %>%
  unite("location", CHROM:POS, sep = ".", remove = F)

sample7 <- read.table("tables_haplotypes/sample7_filtered.haplotypes.table", header = T) %>%
  separate("sample7.AD", c("RC", "AC_1", "AC_2"), 
           sep = ",", convert = T, fill = "right") %>%
  mutate(AC_2_0 = replace(AC_2, is.na(AC_2), 0), .keep = "unused") %>%
  mutate(AC = AC_1 + AC_2_0, .keep = "unused") %>%
  mutate(TC = RC + AC, percentAC = AC/TC) %>%
  mutate(compartment = "soma", genotype = "het", replicate = 4, sample = 7,
         batch = 7, CD1_supp = F, percentCD1Mass = 0) %>%
  unite("location", CHROM:POS, sep = ".", remove = F)

sample8 <- read.table("tables_haplotypes/sample8_filtered.haplotypes.table", header = T) %>%
  separate("sample8.AD", c("RC", "AC_1", "AC_2"), 
           sep = ",", convert = T, fill = "right") %>%
  mutate(AC_2_0 = replace(AC_2, is.na(AC_2), 0), .keep = "unused") %>%
  mutate(AC = AC_1 + AC_2_0, .keep = "unused") %>%
  mutate(TC = RC + AC, percentAC = AC/TC) %>%
  mutate(compartment = "soma", genotype = "null", replicate = 1, sample = 8,
         batch = 8, CD1_supp = F, percentCD1Mass = 0) %>%
  unite("location", CHROM:POS, sep = ".", remove = F)

sample9 <- read.table("tables_haplotypes/sample9_filtered.haplotypes.table", header = T) %>%
  separate("sample9.AD", c("RC", "AC_1", "AC_2"), 
           sep = ",", convert = T, fill = "right") %>%
  mutate(AC_2_0 = replace(AC_2, is.na(AC_2), 0), .keep = "unused") %>%
  mutate(AC = AC_1 + AC_2_0, .keep = "unused") %>%
  mutate(TC = RC + AC, percentAC = AC/TC) %>%
  mutate(compartment = "soma", genotype = "null", replicate = 2, sample = 9,
         batch = 9, CD1_supp = F, percentCD1Mass = 0) %>%
  unite("location", CHROM:POS, sep = ".", remove = F)

sample10 <- read.table("tables_haplotypes/sample10_filtered.haplotypes.table", header = T) %>%
  separate("sample10.AD", c("RC", "AC_1", "AC_2"), 
           sep = ",", convert = T, fill = "right") %>%
  mutate(AC_2_0 = replace(AC_2, is.na(AC_2), 0), .keep = "unused") %>%
  mutate(AC = AC_1 + AC_2_0, .keep = "unused") %>%
  mutate(TC = RC + AC, percentAC = AC/TC) %>%
  mutate(compartment = "soma", genotype = "null", replicate = 3, sample = 10,
         batch = 10, CD1_supp = F, percentCD1Mass = 0) %>%
  unite("location", CHROM:POS, sep = ".", remove = F)

sample11 <- read.table("tables_haplotypes/sample11_filtered.haplotypes.table", header = T) %>%
  separate("sample11.AD", c("RC", "AC_1", "AC_2"), 
           sep = ",", convert = T, fill = "right") %>%
  mutate(AC_2_0 = replace(AC_2, is.na(AC_2), 0), .keep = "unused") %>%
  mutate(AC = AC_1 + AC_2_0, .keep = "unused") %>%
  mutate(TC = RC + AC, percentAC = AC/TC) %>%
  mutate(compartment = "soma", genotype = "null", replicate = 4, sample = 11,
         batch = 11, CD1_supp = F, percentCD1Mass = 0) %>%
  unite("location", CHROM:POS, sep = ".", remove = F)

sample12 <- read.table("tables_haplotypes/sample12_filtered.haplotypes.table", header = T) %>%
  separate("sample12.AD", c("RC", "AC_1", "AC_2"), 
           sep = ",", convert = T, fill = "right") %>%
  mutate(AC_2_0 = replace(AC_2, is.na(AC_2), 0), .keep = "unused") %>%
  mutate(AC = AC_1 + AC_2_0, .keep = "unused") %>%
  mutate(TC = RC + AC, percentAC = AC/TC) %>%
  mutate(compartment = "gc", genotype = "wt", replicate = 1, sample = 12,
         batch = 1, percentLabeledMass = (11*50) / (11*50 + 4*75),
         percentCD1Mass = 1 - percentLabeledMass, CD1_supp = T) %>%
  unite("location", CHROM:POS, sep = ".", remove = F)

sample13 <- read.table("tables_haplotypes/sample13_filtered.haplotypes.table", header = T) %>%
  separate("sample13.AD", c("RC", "AC_1", "AC_2"), 
           sep = ",", convert = T, fill = "right") %>%
  mutate(AC_2_0 = replace(AC_2, is.na(AC_2), 0), .keep = "unused") %>%
  mutate(AC = AC_1 + AC_2_0, .keep = "unused") %>%
  mutate(TC = RC + AC, percentAC = AC/TC) %>%
  mutate(compartment = "gc", genotype = "wt", replicate = 2, sample = 13,
         batch = 2, percentLabeledMass = (11*50) / (11*50 + 4*75),
         percentCD1Mass = 1 - percentLabeledMass, CD1_supp = T) %>%
  unite("location", CHROM:POS, sep = ".", remove = F)

sample14 <- read.table("tables_haplotypes/sample14_filtered.haplotypes.table", header = T) %>%
  separate("sample14.AD", c("RC", "AC_1", "AC_2"), 
           sep = ",", convert = T, fill = "right") %>%
  mutate(AC_2_0 = replace(AC_2, is.na(AC_2), 0), .keep = "unused") %>%
  mutate(AC = AC_1 + AC_2_0, .keep = "unused") %>%
  mutate(TC = RC + AC, percentAC = AC/TC) %>%
  mutate(compartment = "gc", genotype = "wt", replicate = 3, sample = 14,
         batch = 3, percentLabeledMass = (9*50) / (9*50 + 8*75),
         percentCD1Mass = 0, CD1_supp = F) %>%
  unite("location", CHROM:POS, sep = ".", remove = F)

sample15 <- read.table("tables_haplotypes/sample15_filtered.haplotypes.table", header = T) %>%
  separate("sample15.AD", c("RC", "AC_1", "AC_2"), 
           sep = ",", convert = T, fill = "right") %>%
  mutate(AC_2_0 = replace(AC_2, is.na(AC_2), 0), .keep = "unused") %>%
  mutate(AC = AC_1 + AC_2_0, .keep = "unused") %>%
  mutate(TC = RC + AC, percentAC = AC/TC) %>%
  mutate(compartment = "gc", genotype = "het", replicate = 1, sample = 15,
         batch = 4, percentLabeledMass = (11*50) / (11*50 + 11*75),
         percentCD1Mass = 0, CD1_supp = F) %>%
  unite("location", CHROM:POS, sep = ".", remove = F)

sample16 <- read.table("tables_haplotypes/sample16_filtered.haplotypes.table", header = T) %>%
  separate("sample16.AD", c("RC", "AC_1", "AC_2"), 
           sep = ",", convert = T, fill = "right") %>%
  mutate(AC_2_0 = replace(AC_2, is.na(AC_2), 0), .keep = "unused") %>%
  mutate(AC = AC_1 + AC_2_0, .keep = "unused") %>%
  mutate(TC = RC + AC, percentAC = AC/TC) %>%
  mutate(compartment = "gc", genotype = "het", replicate = 2, sample = 16,
         batch = 5, percentLabeledMass = (11*50) / (11*50 + 11*75),
         percentCD1Mass = 0, CD1_supp = F) %>%
  unite("location", CHROM:POS, sep = ".", remove = F)

sample17 <- read.table("tables_haplotypes/sample17_filtered.haplotypes.table", header = T) %>%
  separate("sample17.AD", c("RC", "AC_1", "AC_2"), 
           sep = ",", convert = T, fill = "right") %>%
  mutate(AC_2_0 = replace(AC_2, is.na(AC_2), 0), .keep = "unused") %>%
  mutate(AC = AC_1 + AC_2_0, .keep = "unused") %>%
  mutate(TC = RC + AC, percentAC = AC/TC) %>%
  mutate(compartment = "gc", genotype = "het", replicate = 3, sample = 17,
         batch = 6, percentLabeledMass = (11*50) / (11*50 + 10*75),
         percentCD1Mass = 0, CD1_supp = F) %>%
  unite("location", CHROM:POS, sep = ".", remove = F)

sample18 <- read.table("tables_haplotypes/sample18_filtered.haplotypes.table", header = T) %>%
  separate("sample18.AD", c("RC", "AC_1", "AC_2"), 
           sep = ",", convert = T, fill = "right") %>%
  mutate(AC_2_0 = replace(AC_2, is.na(AC_2), 0), .keep = "unused") %>%
  mutate(AC = AC_1 + AC_2_0, .keep = "unused") %>%
  mutate(TC = RC + AC, percentAC = AC/TC) %>%
  mutate(compartment = "gc", genotype = "het", replicate = 4, sample = 18,
         batch = 7, percentLabeledMass = (10*50) / (10*50 + 11*75),
         percentCD1Mass = 0, CD1_supp = F) %>%
  unite("location", CHROM:POS, sep = ".", remove = F)

sample19 <- read.table("tables_haplotypes/sample19_filtered.haplotypes.table", header = T) %>%
  separate("sample19.AD", c("RC", "AC_1", "AC_2"), 
           sep = ",", convert = T, fill = "right") %>%
  mutate(AC_2_0 = replace(AC_2, is.na(AC_2), 0), .keep = "unused") %>%
  mutate(AC = AC_1 + AC_2_0, .keep = "unused") %>%
  mutate(TC = RC + AC, percentAC = AC/TC) %>%
  mutate(compartment = "gc", genotype = "null", replicate = 1, sample = 19,
         batch = 8, percentLabeledMass = (10*50) / (10*50 + 2*75),
         percentCD1Mass = 1 - percentLabeledMass, CD1_supp = T) %>%
  unite("location", CHROM:POS, sep = ".", remove = F)

sample20 <- read.table("tables_haplotypes/sample20_filtered.haplotypes.table", header = T) %>%
  separate("sample20.AD", c("RC", "AC_1", "AC_2"), 
           sep = ",", convert = T, fill = "right") %>%
  mutate(AC_2_0 = replace(AC_2, is.na(AC_2), 0), .keep = "unused") %>%
  mutate(AC = AC_1 + AC_2_0, .keep = "unused") %>%
  mutate(TC = RC + AC, percentAC = AC/TC) %>%
  mutate(compartment = "gc", genotype = "null", replicate = 2, sample = 20,
         batch = 9, percentLabeledMass = (8*50) / (8*50 + 6*75),
         percentCD1Mass = 1 - percentLabeledMass, CD1_supp = T) %>%
  unite("location", CHROM:POS, sep = ".", remove = F)

sample21 <- read.table("tables_haplotypes/sample21_filtered.haplotypes.table", header = T) %>%
  separate("sample21.AD", c("RC", "AC_1", "AC_2"), 
           sep = ",", convert = T, fill = "right") %>%
  mutate(AC_2_0 = replace(AC_2, is.na(AC_2), 0), .keep = "unused") %>%
  mutate(AC = AC_1 + AC_2_0, .keep = "unused") %>%
  mutate(TC = RC + AC, percentAC = AC/TC) %>%
  mutate(compartment = "gc", genotype = "null", replicate = 3, sample = 21,
         batch = 10, percentLabeledMass = (8*50) / (8*50 + 6*75),
         percentCD1Mass = 1 - percentLabeledMass, CD1_supp = T) %>%
  unite("location", CHROM:POS, sep = ".", remove = F)

sample22 <- read.table("tables_haplotypes/sample22_filtered.haplotypes.table", header = T) %>%
  separate("sample22.AD", c("RC", "AC_1", "AC_2"), 
           sep = ",", convert = T, fill = "right") %>%
  mutate(AC_2_0 = replace(AC_2, is.na(AC_2), 0), .keep = "unused") %>%
  mutate(AC = AC_1 + AC_2_0, .keep = "unused") %>%
  mutate(TC = RC + AC, percentAC = AC/TC) %>%
  mutate(compartment = "gc", genotype = "null", replicate = 4, sample = 22,
         batch = 11, percentLabeledMass = (7*50) / (7*50 + 6*75),
         percentCD1Mass = 1 - percentLabeledMass, CD1_supp = T) %>%
  unite("location", CHROM:POS, sep = ".", remove = F)

# combining samples into a big dataframe while removing gross chromosomes
samples <- bind_rows(sample1, sample2, sample3, sample4, sample5, sample6,
                     sample7, sample8, sample9, sample10, sample11, sample12,
                     sample13, sample14, sample15, sample16, sample17, sample18,
                     sample19, sample20, sample21, sample22) %>%
  filter(CHROM != "JH584304.1", CHROM != "JH584295.1")

# replacing chromsome name for downstream compatibility
samples$location <- str_replace(samples$location, "MT", "M")

# creating a metadata df
metadata_samples <- samples %>%
  group_by(sample) %>%
  summarize(comp = unique(compartment), 
            geno = unique(genotype), 
            rep = unique(replicate))
```

## Visualizing All SNPs

```{r vizAllSNPs}
# distribution of total read counts for SNPs
ggplot(samples, aes(x = TC)) + geom_density(color = "white", size = 1) + 
  boldBlack_theme() + scale_x_continuous(trans = "log10") + 
  labs(x = "Total Read Counts") + 
  geom_vline(xintercept = 10, color = "white", linetype = "dashed")

# setting a read filter of > 9 total counts
high_samples <- samples %>%  
  filter(TC > 9)

# gathering SNP information
snp_number <- high_samples %>%
  group_by(sample) %>%
  summarize(snp_number = n(),
            compartment = unique(compartment),
            genotype = unique(genotype),
            percentCD1Mass = unique(percentCD1Mass),
            percentLabeledMass = unique(percentLabeledMass))

# number of SNPs across samples (higher in CD1-supplemented samples)
ggplot(snp_number, 
       aes(x = sample, y = snp_number, 
           color = compartment)) + 
  geom_point() + boldBlack_theme(base_size = 16) + 
  scale_color_manual(values = cbPalette) +
  labs(x = "Sample Number", y = "SNP Number") +
  theme(legend.position = c(0.15, 0.85))

# number of SNPs directly correlates with ratio of CD1 supplementation
ggplot(snp_number, aes(x = 100*percentCD1Mass, y = snp_number,
                       color = compartment)) +
  scale_color_manual(values = cbPalette) +
  geom_point() + boldBlack_theme(base_size = 16) +
  geom_smooth(method='lm', formula = 'y ~ x', color = "white") +
  stat_regline_equation(label.y = 19000, color = "white", 
                        aes(label = ..eq.label..)) +
  stat_regline_equation(label.y = 18000, color = "white",
                        aes(label = ..rr.label..)) +
  labs(x = "% CD1 Mass", y = "SNP Number") +
  theme(legend.position = c(0.15, 0.85))
```

## Removing B6 SNPs from CD1-supplemented samples

```{r B6SNPremoval}
# grouping samples by SNP location and finding average penetrance
pre_adj_samples <- high_samples %>%
  pivot_wider(names_from = sample, values_from = percentAC) %>%
  group_by(location) %>%
    summarize(mean_noSupp = mean(c(`1`, `2`, `3`, `4`, `5`, `6`, `7`, `8`, 
                                 `9`, `10`, `11`, `14`, `15`, `16`, `17`, 
                                 `18`), na.rm = T),
            mean_Supp = mean(c(`12`, `13`, `19`, `20`, `21`, `22`),
                             na.rm = T),
              presence = case_when(is.na(mean_noSupp) == F & 
                                     is.na(mean_Supp) == F ~ "both",
                                   is.na(mean_noSupp) == T & 
                                     is.na(mean_Supp) == F ~ "supp_only",
                                   is.na(mean_noSupp) == F &
                                     is.na(mean_Supp) == T ~ "noSupp_only"))

# finding effect of supplementation on SNP penetrance
pre_adj_samples[is.na(pre_adj_samples) == T] <- 0
pre_adj_samples_noZeroes <- pre_adj_samples %>%
  mutate(supp_diff = mean_Supp - mean_noSupp)
ggplot(pre_adj_samples_noZeroes, aes(x = supp_diff, color = presence)) + 
  geom_density(size = 1) + boldBlack_theme() + 
  scale_color_manual(values = cbPalette) + 
  labs(x = expression(bold(Delta*"SNP Penetrance (CD1 Supp - No CD1 Supp)")))

# summarizing SNP counts based on supplement status
pre_count_adj_samples <- pre_adj_samples_noZeroes %>%
  group_by(presence) %>%
  summarize(number = n())
pre_count_adj_samples

# only keeping SNPs that are not identified without CD1 supplement
keep_snps <- pre_adj_samples_noZeroes %>%
  filter(presence == "supp_only")
adj_samples <- high_samples %>%
  filter(CD1_supp == T, location %in% keep_snps$location)
```

## Visualizing CD1 SNPs

```{r vizCD1SNPs}
# gathering information for true SNPs
adj_snp_number <- adj_samples %>%
  group_by(sample) %>%
  summarize(adj_snp_number = n(),
            compartment = unique(compartment),
            genotype = unique(genotype),
            percentCD1Mass = unique(percentCD1Mass),
            percentLabeledMass = unique(percentLabeledMass),
            mean_percentAC = mean(percentAC),
            med_percentAC = median(percentAC))

# including zeroes for non-snp samples
adj_snp_number_all <- data.frame(sample = c(1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11,
                                            12, 13, 14, 15, 16, 17, 18, 19, 20,
                                            21, 22),
                                 adj_snp_number = c(0, 0, 0, 0, 0, 0, 0, 0, 0,
                                                    0, 0, 16539, 15248, 0, 0,
                                                    0, 0, 0, 10210, 17185, 
                                                    23373, 19513),
                                 genotype = c("wt", "wt", "wt",
                                              "het", "het", "het", "het",
                                              "null", "null", "null", "null",
                                              "wt", "wt", "wt",
                                              "het", "het", "het", "het",
                                              "null", "null", "null", "null"),
                                 percentCD1Mass = c(0, 0, 0, 0, 0, 0, 0, 0, 0,
                                                    0, 0, 0.3529, 0.3529, 0, 0,
                                                    0, 0, 0, 0.2308, 0.5294, 
                                                    0.5294, 0.5625),
                                 compartment = c("soma", "soma", "soma", "soma",
                                                 "soma", "soma", "soma", "soma",
                                                 "soma", "soma", "soma", "gc", 
                                                 "gc", "gc", "gc", "gc", "gc",
                                                 "gc", "gc", "gc", "gc", "gc"),
                                 gene_number = c(0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
                                                 2799, 2667, 0, 0, 0, 0, 0, 2466,
                                                 2946, 3562, 3251))

# SNPs are now only present in samples with CD1 supplement
ggplot(adj_snp_number_all, 
       aes(x = sample, y = adj_snp_number, 
           color = genotype, shape = compartment)) + 
  geom_point() + boldBlack_theme(base_size = 16) + 
  scale_color_manual(values = cbPalette) + 
  labs(x = "Sample Number", y = "CD1 SNP Number")

# SNP number still strongly correlates with percentage of CD1 supplement mass
ggplot(adj_snp_number_all, aes(x = percentCD1Mass, y = adj_snp_number)) +
  geom_point(color = "white") + boldBlack_theme() +
  geom_smooth(method='lm', formula = 'y ~ x', color = "white") +
  stat_regline_equation(label.y = 20000, color = "white", 
                        aes(label = ..eq.label..)) +
  stat_regline_equation(label.y = 19000, color = "white",
                        aes(label = ..rr.label..)) +
  labs(x = "CD1 Mass Percentage", y = "CD1 SNP Number")

# CD1 SNPs have varying penetrance, partially due to different mass percentages
ggplot(adj_samples, aes(x = percentAC, color = as.factor(percentCD1Mass))) +
  stat_ecdf(size = 1) + boldBlack_theme() +
  labs(x = "CD1 SNP Penetrance")
```

## Combining SNPs to Genes

```{r snpsTOgenes, include = F}
# loading genes and info for mouse
mouse_genes <- genes(TxDb.Mmusculus.UCSC.mm10.knownGene)
mouse_ensembl <- as.data.frame(org.Mm.egENSEMBL)
mouse_chr <- as.data.frame(org.Mm.egCHR)
mouse_pseudo <- as.data.frame(org.Mm.egGENETYPE)

# loading genes and info for humans (for ASD/ID/UTR database conversion)
human_genes <- genes(TxDb.Hsapiens.UCSC.hg19.knownGene)
human_ensembl <- as.data.frame(org.Hs.egENSEMBL)
human_symbols <- as.data.frame(org.Hs.egSYMBOL) %>%
  full_join(human_ensembl, by = "gene_id") %>%
  rename(sfari_humanID = ensembl_id)
human_mart <-  useEnsembl("ensembl", dataset = "hsapiens_gene_ensembl",
                          host = "https://dec2021.archive.ensembl.org/")
mouse_mart <- useEnsembl("ensembl", dataset = "mmusculus_gene_ensembl",
                         host = "https://dec2021.archive.ensembl.org/")
mouse_human <- getLDS(attributes=c("ensembl_gene_id"),
           filters = "ensembl_gene_id", values = mouse_ensembl$ensembl_id, 
           mart=mouse_mart, attributesL=c("ensembl_gene_id"), 
           martL = human_mart) %>%
  rename(ensembl_id = Gene.stable.ID, sfari_humanID = Gene.stable.ID.1)
mouse_human <- left_join(mouse_human, human_symbols, 
                         by = "sfari_humanID")

# sfari genes from sfari.org on 4-21-22
sfari <- read.csv("sfari_042122.csv") %>%
  left_join(mouse_human, by = "sfari_humanID") %>%
  drop_na(ensembl_id) %>%
  select(-gene_id, -symbol)

# id genes from https://doi.org/10.1016%2Fj.ajhg.2015.11.024
id_genes <- read.csv("id_genes.csv") %>%
  left_join(mouse_human %>%
              rename(id_humanID = sfari_humanID), 
            by = "id_humanID") %>%
  drop_na(ensembl_id) %>%
  select(-gene_id, -symbol)

# mouse gene dataframe and grange
mouse_symbols <- as.data.frame(org.Mm.egSYMBOL) %>%
  full_join(mouse_ensembl, by = "gene_id") %>%
  full_join(mouse_pseudo, by = "gene_id") %>%
  left_join(sfari, by = "ensembl_id") %>%
  left_join(id_genes, by = "ensembl_id")
grange_samples <- adj_samples %>%
  select(-CHROM) %>%
  separate(location, c("chrom", "start"), convert = T) %>%
  mutate(end = start, chrom = paste0('chr', chrom)) %>%
  ungroup() %>%
  makeGRangesFromDataFrame(keep.extra.columns = T) 

# converting variants from position info to gene info
overlap_samples <- mergeByOverlaps(grange_samples, mouse_genes) %>%
  as.list() %>%
  as.data.frame() %>%
  select(-contains("mouse"), -contains("strand"), 
         -contains("end"), -contains("width"), -contains("start")) %>%
  select(contains("grange"), contains("gene_id")) %>%
  rename(grange_samples.gene_id = gene_id)
names(overlap_samples) <- substring(names(overlap_samples), 16)
gene_samples <- overlap_samples %>%
  group_by(sample, gene_id) %>%
  summarize(CHR = unique(seqnames), batch = unique(batch), gene_RC = sum(RC),
            gene_AC = sum(AC), gene_TC = sum(TC), replicate = unique(replicate),
            compartment = unique(compartment), genotype = unique(genotype),
            percentCD1Mass = unique(percentCD1Mass),
            percentLabeledMass = unique(percentLabeledMass)) %>%
  mutate(geneAdjPercentAC = gene_AC/gene_TC) %>%
  left_join(mouse_ensembl, by = "gene_id") %>%
  group_by(sample)
```

## Visualizing Gene Data

```{r vizGenes}
# gathering information for SNP-containing genes
gene_number <- gene_samples %>%
  group_by(sample) %>%
  summarize(gene_number = n(),
            compartment = unique(compartment),
            genotype = unique(genotype),
            percentCD1Mass = unique(percentCD1Mass),
            percentLabeledMass = unique(percentLabeledMass))

# number of SNP-containing genes correlates with mass of CD1 supplement
ggplot(adj_snp_number_all, 
       aes(x = sample, y = gene_number, color = genotype, shape = compartment)) + 
  geom_point() + boldBlack_theme(base_size = 16) + 
  scale_color_manual(values = cbPalette) + 
  labs(y = "CD1 Gene Number", x = "Sample Number")
ggplot(adj_snp_number_all, aes(x = percentCD1Mass, y = gene_number)) + 
  geom_point(color = "white") + boldBlack_theme() +
  geom_smooth(method='lm', formula = 'y ~ x', color = "white") +
  stat_regline_equation(label.y = 3500, color = "white", 
                        aes(label = ..eq.label..)) +
  stat_regline_equation(label.y = 3400, color = "white",
                        aes(label = ..rr.label..)) +
  labs(y = "CD1 Gene Number", x = "CD1 Mass Percentage")

# CD1 genes have varying penetrance, partially due to different mass percentages
ggplot(gene_samples, aes(x = 100*geneAdjPercentAC, 
       color = as.factor(round(100*percentCD1Mass)))) +
  geom_density(size = 1) + boldBlack_theme(base_size = 16) + 
  scale_color_manual(values = cbPalette) +
  guides(color=guide_legend(title="% CD1 Mass")) +
  labs(x = "CD1 SNP Penetrance")

# creating and using a function for rowwise binomial testing of variant penetrance
binom.p <- function(x, n, p){binom.test(x, n, p, alternative="less")$p.value}
gene_samples$binom.pValue <- mapply(binom.p,
                                         gene_samples$gene_AC,
                                         gene_samples$gene_TC,
                                         gene_samples$percentCD1Mass)
gene_samples$p.adjust <- p.adjust(gene_samples$binom.pValue, method = "fdr")
gene_samples <- gene_samples %>%
  mutate(ambience = case_when(binom.pValue < 0.0001 ~ "real",
                              TRUE ~ "ambient"),
         adj_ambience = case_when(p.adjust < 0.0001 ~ "real",
                              TRUE ~ "ambient"))

# exclusion list for genes with "bad" snps in > 4/6 samples
ensembl_percents <- gene_samples %>%
  pivot_wider(names_from = sample, values_from = p.adjust,
              names_prefix = "number_") %>%
  group_by(ensembl_id) %>%
  summarize(number_12 = mean(number_12, na.rm = T),
            number_13 = mean(number_13, na.rm = T),
            number_19 = mean(number_19, na.rm = T),
            number_20 = mean(number_20, na.rm = T),
            number_21 = mean(number_21, na.rm = T),
            number_22 = mean(number_22, na.rm = T))
ensembl_percents[ensembl_percents < 0.0001] <- NaN
ensembl_percents$number_good <- rowSums(is.na(ensembl_percents))
ensembl_percents$number_bad <- 6 - ensembl_percents$number_good
ensembl_percents <- filter(ensembl_percents, number_bad > 2)
```

## Loading Counts Data

```{r loadCounts, include = F}
sample1_counts <- read.table("tables_counts/sample1_counts.txt", 
                             header = T) %>%
  select(contains("Geneid"), contains("sample")) %>%
  rename(sample_1 = sample1_recalibrated.readGroups.split.markedDupes.Aligned.sortedByCoord.out.bam)

sample2_counts <- read.table("tables_counts/sample2_counts.txt", 
                             header = T) %>%
  select(contains("Geneid"), contains("sample")) %>%
  rename(sample_2 = sample2_recalibrated.readGroups.split.markedDupes.Aligned.sortedByCoord.out.bam)

sample3_counts <- read.table("tables_counts/sample3_counts.txt", 
                             header = T) %>%
  select(contains("Geneid"), contains("sample")) %>%
  rename(sample_3 = sample3_recalibrated.readGroups.split.markedDupes.Aligned.sortedByCoord.out.bam)

sample4_counts <- read.table("tables_counts/sample4_counts.txt", 
                             header = T) %>%
  select(contains("Geneid"), contains("sample")) %>%
  rename(sample_4 = sample4_recalibrated.readGroups.split.markedDupes.Aligned.sortedByCoord.out.bam)

sample5_counts <- read.table("tables_counts/sample5_counts.txt", 
                             header = T) %>%
  select(contains("Geneid"), contains("sample")) %>%
  rename(sample_5 = sample5_recalibrated.readGroups.split.markedDupes.Aligned.sortedByCoord.out.bam)

sample6_counts <- read.table("tables_counts/sample6_counts.txt", 
                             header = T) %>%
  select(contains("Geneid"), contains("sample")) %>%
  rename(sample_6 = sample6_recalibrated.readGroups.split.markedDupes.Aligned.sortedByCoord.out.bam)

sample7_counts <- read.table("tables_counts/sample7_counts.txt", 
                             header = T) %>%
  select(contains("Geneid"), contains("sample")) %>%
  rename(sample_7 = sample7_recalibrated.readGroups.split.markedDupes.Aligned.sortedByCoord.out.bam)

sample8_counts <- read.table("tables_counts/sample8_counts.txt", 
                             header = T) %>%
  select(contains("Geneid"), contains("sample")) %>%
  rename(sample_8 = sample8_recalibrated.readGroups.split.markedDupes.Aligned.sortedByCoord.out.bam)

sample9_counts <- read.table("tables_counts/sample9_counts.txt", 
                             header = T) %>%
  select(contains("Geneid"), contains("sample")) %>%
  rename(sample_9 = sample9_recalibrated.readGroups.split.markedDupes.Aligned.sortedByCoord.out.bam)

sample10_counts <- read.table("tables_counts/sample10_counts.txt", 
                             header = T) %>%
  select(contains("Geneid"), contains("sample")) %>%
  rename(sample_10 = sample10_recalibrated.readGroups.split.markedDupes.Aligned.sortedByCoord.out.bam)

sample11_counts <- read.table("tables_counts/sample11_counts.txt", 
                             header = T) %>%
  select(contains("Geneid"), contains("sample")) %>%
  rename(sample_11 = sample11_recalibrated.readGroups.split.markedDupes.Aligned.sortedByCoord.out.bam)

sample12_counts <- read.table("tables_counts/sample12_counts.txt", 
                             header = T) %>%
  select(contains("Geneid"), contains("sample")) %>%
  rename(sample_12 = sample12_recalibrated.readGroups.split.markedDupes.Aligned.sortedByCoord.out.bam)

sample13_counts <- read.table("tables_counts/sample13_counts.txt", 
                             header = T) %>%
  select(contains("Geneid"), contains("sample")) %>%
  rename(sample_13 = sample13_recalibrated.readGroups.split.markedDupes.Aligned.sortedByCoord.out.bam)

sample14_counts <- read.table("tables_counts/sample14_counts.txt", 
                             header = T) %>%
  select(contains("Geneid"), contains("sample")) %>%
  rename(sample_14 = sample14_recalibrated.readGroups.split.markedDupes.Aligned.sortedByCoord.out.bam)

sample15_counts <- read.table("tables_counts/sample15_counts.txt", 
                             header = T) %>%
  select(contains("Geneid"), contains("sample")) %>%
  rename(sample_15 = sample15_recalibrated.readGroups.split.markedDupes.Aligned.sortedByCoord.out.bam)

sample16_counts <- read.table("tables_counts/sample16_counts.txt", 
                             header = T) %>%
  select(contains("Geneid"), contains("sample")) %>%
  rename(sample_16 = sample16_recalibrated.readGroups.split.markedDupes.Aligned.sortedByCoord.out.bam)

sample17_counts <- read.table("tables_counts/sample17_counts.txt", 
                             header = T) %>%
  select(contains("Geneid"), contains("sample")) %>%
  rename(sample_17 = sample17_recalibrated.readGroups.split.markedDupes.Aligned.sortedByCoord.out.bam)

sample18_counts <- read.table("tables_counts/sample18_counts.txt", 
                             header = T) %>%
  select(contains("Geneid"), contains("sample")) %>%
  rename(sample_18 = sample18_recalibrated.readGroups.split.markedDupes.Aligned.sortedByCoord.out.bam)

sample19_counts <- read.table("tables_counts/sample19_counts.txt", 
                             header = T) %>%
  select(contains("Geneid"), contains("sample")) %>%
  rename(sample_19 = sample19_recalibrated.readGroups.split.markedDupes.Aligned.sortedByCoord.out.bam)

sample20_counts <- read.table("tables_counts/sample20_counts.txt", 
                             header = T) %>%
  select(contains("Geneid"), contains("sample")) %>%
  rename(sample_20 = sample20_recalibrated.readGroups.split.markedDupes.Aligned.sortedByCoord.out.bam)

sample21_counts <- read.table("tables_counts/sample21_counts.txt", 
                             header = T) %>%
  select(contains("Geneid"), contains("sample")) %>%
  rename(sample_21 = sample21_recalibrated.readGroups.split.markedDupes.Aligned.sortedByCoord.out.bam)

sample22_counts <- read.table("tables_counts/sample22_counts.txt", 
                             header = T) %>%
  select(contains("Geneid"), contains("sample")) %>%
  rename(sample_22 = sample22_recalibrated.readGroups.split.markedDupes.Aligned.sortedByCoord.out.bam)

# combining counts data into a single dataframe
counts_samples <- bind_cols(sample1_counts, sample2_counts, sample3_counts,
                            sample4_counts, sample5_counts, sample6_counts,
                            sample7_counts, sample8_counts, sample9_counts,
                            sample10_counts, sample11_counts, sample12_counts,
                            sample13_counts, sample14_counts, sample15_counts,
                            sample16_counts, sample17_counts, sample18_counts,
                            sample19_counts, sample20_counts, sample21_counts,
                            sample22_counts) %>%
  select(Geneid...1, contains("sample")) %>%
  rename(ensembl_id = Geneid...1)
```

## DESeq2: WT Somata vs WT GCs

```{r somaGCs, echo = F}
# setting up and filtering count matrix
all_na_counts_samples <- counts_samples %>%
  select(ensembl_id, sample_1:sample_22) %>%
  filter(!(ensembl_id %in% ensembl_percents$ensembl_id)) 
all_na_counts_samples[all_na_counts_samples == 0] <- NA
all_filtered_samples <- all_na_counts_samples %>%
  filter(!if_all(.cols = sample_1:sample_3, .fns = is.na) |
           !if_all(.cols = sample_4:sample_7, .fns = is.na) |
           !if_all(.cols = sample_8:sample_11, .fns = is.na) |
           !if_all(.cols = sample_12:sample_14, .fns = is.na) |
           !if_all(.cols = sample_15:sample_18, .fns = is.na) |
           !if_all(.cols = sample_19:sample_22, .fns = is.na)
           ) %>%
  column_to_rownames(var = "ensembl_id")
all_filtered_samples[is.na(all_filtered_samples)] <- 0

# setting up metadata matrix
all_metadata_samples <- metadata_samples %>%
  unite(col = "combo", comp, geno)
rownames(all_metadata_samples) <- colnames(all_filtered_samples)

# running DESeq2
all_dds <- DESeqDataSetFromMatrix(countData = all_filtered_samples,
                                  colData = all_metadata_samples,
                                  design = ~ combo)
all_dds$combo <- relevel(all_dds$combo, ref = "gc_wt")
all_dds <- DESeq(all_dds)
resultsNames(all_dds)

# loose filter to remove junk
all_dds <- all_dds[rowSums(counts(all_dds)) >= 10,]

# shrinking and storing DESeq2 results
all_res_shrink <- lfcShrink(all_dds, 
                                    coef = "combo_soma_wt_vs_gc_wt", 
                            type = "apeglm")
summary(all_res_shrink, alpha = 0.05)
allShrink_plot <- as.data.frame(all_res_shrink) %>%
  drop_na() %>%
  rownames_to_column("ensembl_id") %>%
  left_join(mouse_symbols, by = "ensembl_id") %>%
  mutate(significant = case_when(padj < 0.05 ~ T, padj >= 0.05 ~ F),
         direction = case_when(log2FoldChange < 0 ~ "gc",
                               log2FoldChange > 0 ~ "soma"),
         log2FoldChange = -log2FoldChange,
         label = case_when(padj < 0.05 ~ symbol, padj >= 0.05 ~ "")) %>%
  dplyr::select(-contains("sfari"))

# transforming for visualization
all_rld_blind <- rlog(all_dds, blind = T)

# heatmap of the count matrix
all_select <- order(rowMeans(counts(all_dds, normalized = TRUE)),
                decreasing = TRUE)[1:20]
all_df <- as.data.frame(colData(all_dds)[,"combo"])
pheatmap(assay(all_rld_blind)[all_select,], cluster_rows = F, show_rownames = F,
         cluster_cols = T)

# heatmap of sample-to-sample distances
all_sampleDists <- dist(t(assay(all_rld_blind)))
all_sampleDistMatrix <- as.matrix(all_sampleDists)
rownames(all_sampleDistMatrix) <- paste(all_rld_blind$combo, 
                                        all_rld_blind$rep, sep = "-")
colnames(all_sampleDistMatrix) <- paste(all_rld_blind$combo,
                                               all_rld_blind$rep, 
                                               sep = "-")
all_colors <- colorRampPalette(brewer.pal(9, "RdYlBu"))(255)

# principal component plot of the samples
plotPCA(all_rld_blind, intgroup = "combo") +
  theme_classic(base_size = 8, base_family = "sans") + 
  theme(legend.position = c(0.8, 0.8))

# visualize optimization of independent filtering to maximize number of hits
plot(metadata(all_res_shrink)$filterNumRej,
     type = "b", ylab = "number of rejections",
     xlab = "quantiles of filter")
lines(metadata(all_res_shrink)$lo.fit, col = "red")
abline(v = metadata(all_res_shrink)$filterTheta)

# fig 3, supp fig 2c: gc vs soma localization
real_fig_3_supp_2c <- ggplot(allShrink_plot %>%
                               filter(gene_type == "protein-coding") %>%
                          mutate(localization = case_when(
                            log2FoldChange > 0 & significant == T ~ "GC",
                            log2FoldChange < 0 & significant == T ~ "Soma",
                            TRUE ~ "None"
                          )),
       aes(x = log10(baseMean), y = log2FoldChange, color = localization)) +
  geom_point(size = 0.0000000001) +
  labs(x = bquote(log[10](TPM)),
       y = bquote(log[2](FC)),
       color = bquote(Enrichment)) +
  theme_classic(base_size = 7, base_family = "sans") + 
  scale_color_manual(breaks = c("GC", "None", "Soma"),
                     values = c("#E69F00", "purple", "#56B4E9")) +
    theme(legend.position = c(0.8, 0.9),
          legend.title = element_text(size = 7),
          legend.text = element_text(size = 7),
          legend.background = element_rect(fill=NA),
          legend.spacing.x = unit(0.0002, "in"),
          legend.spacing.y = unit(0.01, "in"),
          axis.text.x = element_text(color = "black"),
          axis.text.y = element_text(color = "black"),
          axis.ticks = element_line(color = "black")); real_fig_3_supp_2c
ggsave("real_fig_3_supp_2c.pdf", plot = real_fig_3_supp_2c, units = "in",
       height = 1.75, width = 1.75) # 1 inch = 72 pt

# defining subcellular localization for downstream processes
subcell <- allShrink_plot %>%
  rename(subcell = log2FoldChange) %>%
  mutate(localization = case_when(subcell > 0 & significant == T ~ "GC",
                                  subcell < 0 & significant == T ~ "Soma",
                                  TRUE ~ "None")) %>%
  select(ensembl_id, subcell, localization)
```

## DESeq2: Somata With Heterozygotes

```{r somaHet, echo = F}
# setting up and filtering count matrix
somata_het_na_counts_samples <- counts_samples %>%
  select(ensembl_id, sample_1:sample_11)
somata_het_na_counts_samples[somata_het_na_counts_samples == 0] <- NA
somata_het_filtered_samples <- somata_het_na_counts_samples %>%
  filter(!if_all(.cols = sample_1:sample_3, .fns = is.na) |
           !if_all(.cols = sample_4:sample_7, .fns = is.na) |
           !if_all(.cols = sample_8:sample_11, .fns = is.na)) %>%
  column_to_rownames(var = "ensembl_id")
somata_het_filtered_samples[is.na(somata_het_filtered_samples)] <- 0

# setting up metadata matrix
somata_het_metadata_samples <- metadata_samples %>%
  filter(comp == "soma")
rownames(somata_het_metadata_samples) <- colnames(somata_het_filtered_samples)

# running DESeq2
somata_het_dds <- DESeqDataSetFromMatrix(countData = somata_het_filtered_samples,
                                  colData = somata_het_metadata_samples,
                                  design = ~ geno)
somata_het_dds$geno <- relevel(somata_het_dds$geno, ref = "wt")
somata_het_dds <- DESeq(somata_het_dds)
resultsNames(somata_het_dds)

# loose filter to remove noise
somata_het_dds <- somata_het_dds[rowSums(counts(somata_het_dds)) >= 10,]

# shrinking and storing DESeq2 results
somata_het_res_shrink <- lfcShrink(somata_het_dds, 
                                    coef = "geno_null_vs_wt", type = "apeglm")
summary(somata_het_res_shrink, alpha = 0.05)
somata_hetShrink_plot <- as.data.frame(somata_het_res_shrink) %>%
  drop_na() %>%
  rownames_to_column("ensembl_id") %>%
  left_join(mouse_symbols, by = "ensembl_id") %>%
  inner_join(subcell, by = "ensembl_id") %>%
  mutate(significant = case_when(padj < 0.05 ~ T, padj >= 0.05 ~ F),
         direction = case_when(log2FoldChange < 0 ~ "Depleted",
                               log2FoldChange > 0 ~ "Enriched"),
         label = case_when(padj < 0.05 ~ symbol, padj >= 0.05 ~ "")) %>%
  group_by(ensembl_id) %>%
  slice(1L) %>%
  ungroup()

# transforming values for visualization
somata_het_rld_blind <- rlog(somata_het_dds, blind = T)

# heatmap of the count matrix
somata_het_select <- order(rowMeans(counts(somata_het_dds, normalized = TRUE)),
                decreasing = TRUE)[1:20]
somata_het_df <- as.data.frame(colData(somata_het_dds)[,"geno"])
pheatmap(assay(somata_het_rld_blind)[somata_het_select,], cluster_rows = F, show_rownames = F,
         cluster_cols = T)

# heatmap of sample-to-sample distances
somata_het_sampleDists <- dist(t(assay(somata_het_rld_blind)))
somata_het_sampleDistMatrix <- as.matrix(somata_het_sampleDists)
rownames(somata_het_sampleDistMatrix) <- paste(somata_het_rld_blind$comp,
                                               somata_het_rld_blind$geno, 
                                        somata_het_rld_blind$rep, sep = "-")
colnames(somata_het_sampleDistMatrix) <- paste(somata_het_rld_blind$comp,
                                               somata_het_rld_blind$geno,
                                               somata_het_rld_blind$rep, 
                                               sep = "-")
somata_het_colors <- colorRampPalette(brewer.pal(9, "RdYlBu"))(255)

# fig 3, supp fig 2 b: soma clustering
real_fig_3_supp_2b <- pheatmap(somata_het_sampleDistMatrix,
         clustering_distance_rows = somata_het_sampleDists,
         clustering_distance_cols = somata_het_sampleDists,
         col = somata_het_colors,
         cellwidth = 5, cellheight = 5, fontsize = 5,
         treeheight_row = 2.5, treeheight_col = 2.5)
real_fig_3_supp_2b$gtable$grobs[[1]]$gp <- gpar(lwd = 0.5)
real_fig_3_supp_2b$gtable$grobs[[2]]$gp <- gpar(lwd = 0.5)
real_fig_3_supp_2b
ggsave("real_fig_3_supp_2b.pdf", plot = real_fig_3_supp_2b, units = "in",
       height = 1.8, width = 1.8) # 1 inch = 72 pt

# principal component plot of the samples
plotPCA(somata_het_rld_blind, intgroup = "geno") +
  geom_text_repel(aes(label = name))

# visualize optimization of independent filtering to maximize number of hits
plot(metadata(somata_het_res_shrink)$filterNumRej,
     type = "b", ylab = "number of rejections",
     xlab = "quantiles of filter")
lines(metadata(somata_het_res_shrink)$lo.fit, col = "red")
abline(v = metadata(somata_het_res_shrink)$filterTheta)
```

## DESeq2: GCs with Heterozygotes and Excluded SNPs

```{r DESeq2HetExclude, echo = F}
# setting up and filtering count matrix
het_exclude_na_counts_samples <- counts_samples %>%
  select(ensembl_id, sample_12:sample_22) %>%
  filter(!(ensembl_id %in% ensembl_percents$ensembl_id))
het_exclude_na_counts_samples[het_exclude_na_counts_samples == 0] <- NA
het_exclude_filtered_samples <- het_exclude_na_counts_samples %>%
  filter(!if_all(.cols = sample_12:sample_14, .fns = is.na) |
           !if_all(.cols = sample_15:sample_18, .fns = is.na) |
           !if_all(.cols = sample_19:sample_22, .fns = is.na)) %>%
  column_to_rownames(var = "ensembl_id")
het_exclude_filtered_samples[is.na(het_exclude_filtered_samples)] <- 0

# setting up metadata matrix
het_exclude_metadata_samples <- metadata_samples %>%
  filter(comp == "gc")
rownames(het_exclude_metadata_samples) <- colnames(het_exclude_filtered_samples)

# running DESeq2
het_exclude_dds <- DESeqDataSetFromMatrix(countData = het_exclude_filtered_samples,
                                  colData = het_exclude_metadata_samples,
                                  design = ~ geno)
het_exclude_dds$geno <- relevel(het_exclude_dds$geno, ref = "wt")
het_exclude_dds <- DESeq(het_exclude_dds)
resultsNames(het_exclude_dds)

# loose filtering to remove junk
het_exclude_dds <- het_exclude_dds[rowSums(counts(het_exclude_dds)) >= 10,]

# shrinking and storing DESeq2 results
het_exclude_res_shrink <- lfcShrink(het_exclude_dds, 
                                    coef = "geno_null_vs_wt", type = "apeglm")
summary(het_exclude_res_shrink, alpha = 0.05)
het_excludeShrink_plot <- as.data.frame(het_exclude_res_shrink) %>%
  drop_na() %>%
  rownames_to_column("ensembl_id") %>%
  left_join(mouse_symbols, by = "ensembl_id") %>%
  inner_join(subcell, by = "ensembl_id") %>%
  mutate(significant = case_when(padj < 0.05 ~ T, padj >= 0.05 ~ F),
         direction = case_when(log2FoldChange > 0 ~ "Enriched",
                               log2FoldChange < 0 ~ "Depleted"),
         label = case_when(padj < 0.05 ~ symbol, padj >= 0.05 ~ "")) %>%
  group_by(ensembl_id) %>%
  slice(1L) %>%
  ungroup()

# transforming values for visualization
het_exclude_rld_blind <- rlog(het_exclude_dds, blind = T)

# heatmap of the count matrix
het_exclude_select <- order(rowMeans(counts(het_exclude_dds, normalized = TRUE)),
                decreasing = TRUE)[1:20]
het_exclude_df <- as.data.frame(colData(het_exclude_dds)[,"geno"])
pheatmap(assay(het_exclude_rld_blind)[het_exclude_select,], cluster_rows = F, show_rownames = F,
         cluster_cols = T)

# heatmap of sample-to-sample distances
het_exclude_sampleDists <- dist(t(assay(het_exclude_rld_blind)))
het_exclude_sampleDistMatrix <- as.matrix(het_exclude_sampleDists)
rownames(het_exclude_sampleDistMatrix) <- paste(het_exclude_rld_blind$comp,
                                                het_exclude_rld_blind$geno, 
                                        het_exclude_rld_blind$rep, sep = "-")
colnames(het_exclude_sampleDistMatrix) <- paste(het_exclude_rld_blind$comp,
                                                het_exclude_rld_blind$geno, 
                                        het_exclude_rld_blind$rep, sep = "-")
het_exclude_colors <- colorRampPalette(brewer.pal(9, "RdYlBu"))(255)

# fig 3, supp fig 2 a: gc clustering
real_fig_3_supp_2a <- pheatmap(het_exclude_sampleDistMatrix,
         clustering_distance_rows = het_exclude_sampleDists,
         clustering_distance_cols = het_exclude_sampleDists,
         col = het_exclude_colors,
         cellwidth = 5, cellheight = 5, fontsize = 5,
         treeheight_row = 2.5, treeheight_col = 2.5)
real_fig_3_supp_2a$gtable$grobs[[1]]$gp <- gpar(lwd = 0.5)
real_fig_3_supp_2a$gtable$grobs[[2]]$gp <- gpar(lwd = 0.5)
real_fig_3_supp_2a
ggsave("real_fig_3_supp_2a.pdf", plot = real_fig_3_supp_2a, units = "in",
       height = 1.8, width = 1.8) # 1 inch = 72 pt

# principal component plot of the samples
plotPCA(het_exclude_rld_blind, intgroup = "geno") +
  geom_text_repel(aes(label = name))

# visualize optimization of independent filtering to maximize number of hits
plot(metadata(het_exclude_res_shrink)$filterNumRej,
     type = "b", ylab = "number of rejections",
     xlab = "quantiles of filter")
lines(metadata(het_exclude_res_shrink)$lo.fit, col = "red")
abline(v = metadata(het_exclude_res_shrink)$filterTheta)

# dataframe for easy figure creation
fig_3b_labels_colors <- het_excludeShrink_plot %>%
  filter(ensembl_id != "ENSMUSG00000117786") %>%
  mutate(label_keep = case_when(significant == T & 
                                   gene_type == "protein-coding" ~ "yes"),
         color_keep = case_when(significant == T ~ "yes"))
fig_3b_labels_colors$symbol[is.na(fig_3b_labels_colors$label_keep)] <- ""
fig_3b_labels_colors$localization[is.na(fig_3b_labels_colors$color_keep)] <- NA
sig_fig_3b_labels_colors <- fig_3b_labels_colors %>%
  filter(label_keep == "yes")

# figure 3: gc ma plot with colored subcellular localization
real_fig_3b <- ggplot(fig_3b_labels_colors %>%
                        filter(gene_type == "protein-coding"),
       aes(x = log10(baseMean), y = log2FoldChange, color = localization)) +
  geom_point(size = 0.0000000001) +
  labs(x = bquote(log[10](TPM)), # log10(Mean of Normalized Counts)
       y = bquote(log[2](FC)), # Shrunken log2(FC) for GCs After Bcl11a Deletion
       color = bquote(Enrichment)) +
  theme_classic(base_size = 7, base_family = "sans") + 
  geom_text_repel(data = fig_3b_labels_colors %>%
                    filter(symbol == "Pcdhac2"),
                  aes(label = symbol, color = localization),
                  size = 1.25, show.legend = F,
                  box.padding = 1, force = 1, force_pull = 1,
                  point.padding = 0, nudge_x = -0.25, nudge_y = 0.25) +
  geom_text_repel(data = fig_3b_labels_colors %>%
                    filter(symbol == "Mmp24"),
                  aes(label = symbol, color = localization), 
                  size = 1.25, show.legend = F,
                  box.padding = 0.75, force = 1, force_pull = 1,
                  point.padding = 0, nudge_y = -0.25) +
    geom_text_repel(data = fig_3b_labels_colors %>%
                    filter(symbol != "Pcdhac2",
                             symbol != "Mmp24",
                           gene_type == "protein-coding"),
                  aes(label = symbol, color = localization),
                  size = 1.25, show.legend = F,
                  box.padding = 0.125, force = 1, force_pull = 1,
                  point.padding = 0, nudge_x = 0.125, nudge_y = 0) +
  scale_color_manual(breaks = c("GC", "None", "Soma"),
                     values = c("#E69F00", "purple", "#56B4E9"),
                     na.value = "grey20") +
    theme(legend.position = c(0.8, 0.85),
          legend.title = element_text(size = 7),
          legend.text = element_text(size = 7),
          legend.background = element_rect(fill=NA),
          legend.spacing.x = unit(0.0002, "in"),
          legend.spacing.y = unit(0.01, "in"),
          axis.text.x = element_text(color = "black"),
          axis.text.y = element_text(color = "black"),
          axis.ticks = element_line(color = "black")); real_fig_3b
ggsave("real_fig_3b.pdf", plot = real_fig_3b, units = "in",
       height = 1.75, width = 1.75) # 1 inch = 72 pt

# DAGs
het_exclude_sigAll <- het_excludeShrink_plot %>%
  filter(significant == T,
         gene_type == "protein-coding")
het_exclude_sigNull <- het_excludeShrink_plot %>%
  filter(significant == T, direction == "Enriched",
         gene_type == "protein-coding")
het_exclude_sigWT <- het_excludeShrink_plot %>%
  filter(significant == T, direction == "Depleted",
         gene_type == "protein-coding")

# # GO enrichment for GCs (maybe remove?)
# het_exclude_goAll <- enrichGO(gene = het_exclude_sigAll$ensembl_id,
#                              OrgDb = org.Mm.eg.db,
#                              keyType = "ENSEMBL",
#                              ont = "ALL",
#                              pAdjustMethod = "BH",
#                              readable = T,
#                              pool = F,
#                              universe = het_excludeShrink_plot$ensembl_id)
# het_exclude_goNull <- enrichGO(gene = het_exclude_sigNull$ensembl_id,
#                              OrgDb = org.Mm.eg.db,
#                              keyType = "ENSEMBL",
#                              ont = "ALL",
#                              pAdjustMethod = "BH",
#                              readable = T,
#                              pool = F,
#                              universe = het_excludeShrink_plot$ensembl_id)
# het_exclude_goWT <- enrichGO(gene = het_exclude_sigWT$ensembl_id,
#                              OrgDb = org.Mm.eg.db,
#                              keyType = "ENSEMBL",
#                              ont = "ALL",
#                              pAdjustMethod = "BH",
#                              readable = T,
#                              pool = F,
#                              universe = het_excludeShrink_plot$ensembl_id)
# dotplot(het_exclude_goAll) + boldBlack_theme() +
#   ggtitle("GO Enrichment: Excluded GCs, Up/Down in Null")
# dotplot(het_exclude_goNull) + boldBlack_theme() +
#   ggtitle("GO Enrichment: Excluded GCs, Up in Null")
# dotplot(het_exclude_goWT) + boldBlack_theme() +
#   ggtitle("GO Enrichment: Excluded GCs, Down in Null")
# 
# # reducing go term lists based on semantic similarity with revigo
# gcAll_test <- as.data.frame(het_exclude_goAll)
# gcAll_test$pvalue <- format(gcAll_test$pvalue, scientific = F)
# write.table(gcAll_test %>% select(ID, pvalue), file = "gcAll_GO.txt",
#             row.names = F, col.names = F, quote = F)
# gcAll_revi <- read.table("Revigo_gcAllBP.tsv", header = T) %>%
#     select(-contains("PC")) %>%
#   full_join(read.table("Revigo_gcAllCC.tsv", header = T) %>%
#                 select(-contains("PC"))) %>%
#   full_join(read.table("Revigo_gcAllMF.tsv", header = T) %>%
#                 select(-contains("PC"))) %>%
#   filter(Dispensability < 0.7) %>%
#   rename(ID = TermID) %>%
#   left_join(gcAll_test, by = "ID") %>%
#   separate(GeneRatio, into = c("GeneNum", "GeneDenom")) %>%
#   mutate(GeneRatio = as.numeric(GeneNum) / as.numeric(GeneDenom))
# ggplot(gcAll_revi %>%
#        mutate(Description = fct_reorder2(Description,
#                                          GeneRatio,
#                                          -GeneRatio)) %>%
#          slice(1:10), aes(x = GeneRatio,
#                           y = Description,
#                       size = Count,
#                       color = p.adjust)) +
#   geom_point() + theme_classic(base_size = 8, base_family = "sans") +
#   scale_size_continuous(range = c(0.25, 2.5)) + ylab("") +
#     scale_color_distiller(palette = "RdYlBu")
# gcNull_test <- as.data.frame(het_exclude_goNull)
# gcNull_test$pvalue <- format(gcNull_test$pvalue, scientific = F)
# write.table(gcNull_test %>% select(ID, pvalue), file = "gcNull_GO.txt",
#             row.names = F, col.names = F, quote = F)
# gcNull_revi <- read.table("Revigo_gcNullBP.tsv", header = T) %>%
#     select(-contains("PC")) %>%
#   full_join(read.table("Revigo_gcNullCC.tsv", header = T) %>%
#                 select(-contains("PC"))) %>%
#   full_join(read.table("Revigo_gcNullMF.tsv", header = T) %>%
#                 select(-contains("PC"))) %>%
#   filter(Dispensability < 0.7) %>%
#   rename(ID = TermID) %>%
#   left_join(gcNull_test, by = "ID") %>%
#   separate(GeneRatio, into = c("GeneNum", "GeneDenom")) %>%
#   mutate(GeneRatio = as.numeric(GeneNum) / as.numeric(GeneDenom))
# ggplot(gcNull_revi %>%
#        mutate(Description = fct_reorder2(Description,
#                                          GeneRatio,
#                                          -GeneRatio)) %>%
#          slice(1:10), aes(x = GeneRatio,
#                           y = Description,
#                       size = Count,
#                       color = p.adjust)) +
#   geom_point() + theme_classic(base_size = 8, base_family = "sans") +
#   scale_size_continuous(range = c(0.25, 2.5)) + ylab("") +
#     scale_color_distiller(palette = "RdYlBu")
# gcWT_test <- as.data.frame(het_exclude_goWT)
# gcWT_test$pvalue <- format(gcWT_test$pvalue, scientific = F)
# write.table(gcWT_test %>% select(ID, pvalue), file = "gcWT_GO.txt",
#             row.names = F, col.names = F, quote = F)
# gcWT_revi <- read.table("Revigo_gcWTBP.tsv", header = T) %>%
#     select(-contains("PC")) %>%
#   full_join(read.table("Revigo_gcWTCC.tsv", header = T) %>%
#                 select(-contains("PC"))) %>%
#   full_join(read.table("Revigo_gcWTMF.tsv", header = T) %>%
#                 select(-contains("PC"))) %>%
#   filter(Dispensability < 0.7) %>%
#   rename(ID = TermID) %>%
#   left_join(gcWT_test, by = "ID") %>%
#   separate(GeneRatio, into = c("GeneNum", "GeneDenom")) %>%
#   mutate(GeneRatio = as.numeric(GeneNum) / as.numeric(GeneDenom))
# ggplot(gcWT_revi %>%
#        mutate(Description = fct_reorder2(Description,
#                                          GeneRatio,
#                                          -GeneRatio)) %>%
#          slice(1:10), aes(x = GeneRatio,
#                           y = Description,
#                       size = Count,
#                       color = p.adjust)) +
#   geom_point() + theme_classic(base_size = 8, base_family = "sans") +
#   scale_size_continuous(range = c(0.25, 2.5)) + ylab("") +
#     scale_color_distiller(palette = "RdYlBu")

# SFARI enrichment
sfari_enrich <- mouse_symbols %>%
  mutate(term = case_when(is.na(sfari_humanID) ~ "non-sfari",
                          !is.na(sfari_humanID) ~ "sfari")) %>%
  select(term, ensembl_id)
het_exclude_sfariAll <- enricher(gene = het_exclude_sigAll$ensembl_id,
                             universe = het_excludeShrink_plot$ensembl_id,
                             TERM2GENE = sfari_enrich,
                             pvalueCutoff = 1,
                             qvalueCutoff = 1,
                             minGSSize = 0,
                             maxGSSize = 20000)
het_exclude_sfariNull <- enricher(gene = het_exclude_sigNull$ensembl_id,
                             universe = het_excludeShrink_plot$ensembl_id,
                             TERM2GENE = sfari_enrich,
                             pvalueCutoff = 1,
                             qvalueCutoff = 1,
                             minGSSize = 0,
                             maxGSSize = 20000)
het_exclude_sfariWT <- enricher(gene = het_exclude_sigWT$ensembl_id,
                             universe = het_excludeShrink_plot$ensembl_id,
                             TERM2GENE = sfari_enrich,
                             pvalueCutoff = 1,
                             qvalueCutoff = 1,
                             minGSSize = 0,
                             maxGSSize = 20000)
dotplot(het_exclude_sfariAll) + boldBlack_theme() +
  ggtitle("SFARI Enrichment: Excluded GCs, Up/Down in Null")
dotplot(het_exclude_sfariNull) + boldBlack_theme() +
  ggtitle("SFARI Enrichment: Excluded GCs, Up in Null")
dotplot(het_exclude_sfariWT) + boldBlack_theme() +
  ggtitle("SFARI Enrichment: Excluded GCs, Down in Null")

# ID enrichment
id_enrich <- mouse_symbols %>%
  mutate(term = case_when(is.na(id_humanID) ~ "non-id",
                          !is.na(id_humanID) ~ "id")) %>%
  select(term, ensembl_id)
het_exclude_idAll <- enricher(gene = het_exclude_sigAll$ensembl_id,
                             universe = het_excludeShrink_plot$ensembl_id,
                             TERM2GENE = id_enrich,
                             pvalueCutoff = 1,
                             qvalueCutoff = 1,
                             minGSSize = 0,
                             maxGSSize = 20000)
het_exclude_idNull <- enricher(gene = het_exclude_sigNull$ensembl_id,
                             universe = het_excludeShrink_plot$ensembl_id,
                             TERM2GENE = id_enrich,
                             pvalueCutoff = 1,
                             qvalueCutoff = 1,
                             minGSSize = 0,
                             maxGSSize = 20000)
het_exclude_idWT <- enricher(gene = het_exclude_sigWT$ensembl_id,
                             universe = het_excludeShrink_plot$ensembl_id,
                             TERM2GENE = id_enrich,
                             pvalueCutoff = 1,
                             qvalueCutoff = 1,
                             minGSSize = 0,
                             maxGSSize = 20000)
dotplot(het_exclude_idAll) + boldBlack_theme() +
  ggtitle("ID Enrichment: Excluded GCs, Up/Down in Null")
dotplot(het_exclude_idNull) + boldBlack_theme() +
  ggtitle("ID Enrichment: Excluded GCs, Up in Null")
dotplot(het_exclude_idWT) + boldBlack_theme() +
  ggtitle("ID Enrichment: Excluded GCs, Down in Null")

# combining gc and soma data
het_excludeShrink_somata_plot <- het_excludeShrink_plot %>%
  filter(ensembl_id != "ENSMUSG00000117786") %>%
  mutate(label_keep = case_when(significant == T & 
                                   gene_type == "protein-coding" ~ "yes"),
         color_keep = case_when(significant == T ~ "yes")) %>%
  filter(label_keep == "yes") %>%
  left_join(somata_hetShrink_plot, by = "ensembl_id")
  
# table 1: gc and soma differential expression (wt vs null)
write.csv(x = het_excludeShrink_somata_plot, 
          file = "exclude_nullVSwt_062522_pt0001ambient_pt05deseq.csv", 
          row.names = F)

test <- fig_3b_labels_colors %>%
                   left_join(somata_hetShrink_plot, by = "ensembl_id")

# fig 3c: quadrant plot of DAGs colored by SFARI status
real_fig_3c <- ggplot(fig_3b_labels_colors %>%
                   left_join(somata_hetShrink_plot, by = "ensembl_id") %>%
                   filter(label_keep == "yes") %>%
                     filter(significant.y == T), 
       aes(x = log2FoldChange.x, 
           y = log2FoldChange.y,
           color = !(is.na(sfari_humanID.x)))) +
  geom_vline(xintercept = 0, color = "black", 
             size = 0.25, linetype = "dashed") +
  geom_hline(yintercept = 0, color = "black", 
             size = 0.25,linetype = "dashed") +
  scale_color_manual(values = c("#0072B2", "#D55E00")) +
  theme_classic(base_size = 7, base_family = "sans") + 
  theme(legend.position = c(0.775, 0.7),
        legend.background = element_rect(fill = NA),
        legend.title = element_text(size = 7),
          legend.text = element_text(size = 7),
        legend.spacing.x = unit(0.0002, "in"),
          legend.spacing.y = unit(0.01, "in"),
        axis.text.x = element_text(color = "black"),
          axis.text.y = element_text(color = "black"),
          axis.ticks = element_line(color = "black")) +
  guides(color = guide_legend(title = bquote(SFARI~Gene), reverse = T,
                              override.aes = list(size = 0.5))) +
  geom_point(size = 0.00000000000000001)  +
  scale_x_continuous(sec.axis = dup_axis(name = "",
                                         breaks = NULL,
                                         labels = NULL)) +
  scale_y_continuous(sec.axis = dup_axis(name = "",
                                         breaks = NULL,
                                         labels = NULL)) +
  labs(x = bquote(GC~log[2](FC)),
       y = bquote(Soma~log[2](FC))) +
  geom_text_repel(data = fig_3b_labels_colors %>%
                   left_join(somata_hetShrink_plot, by = "ensembl_id") %>%
                   filter(symbol.x == "Pcdhac2"),
                  aes(label = symbol.x), 
                  size = 1.25, show.legend = F,
                  box.padding = 0.1, force = 1, force_pull = 1,
                  point.padding = 0, nudge_y = 0.5, nudge_x = -1,
                  min.segment.length = 0.00000000001) +
    geom_text_repel(data = fig_3b_labels_colors %>%
                   left_join(somata_hetShrink_plot, by = "ensembl_id") %>%
                   filter(symbol.x == "Mmp24"),
                  aes(label = symbol.x), 
                  size = 1.25, show.legend = F,
                  box.padding = 1, force = 1, force_pull = 1,
                  point.padding = 0, nudge_y = 1, nudge_x = 0.125) +
  geom_text_repel(data = fig_3b_labels_colors %>%
                   left_join(somata_hetShrink_plot, by = "ensembl_id") %>%
                   filter(label_keep == "yes",
                          symbol.x != "Pcdhac2",
                          symbol.x != "Mmp24",
                          symbol.x != "CYTB",
                          symbol.x != "Elovl2",
                          symbol.x != "Nr4a1"),
                   aes(label = symbol.x),
                  size = 1.25, show.legend = F,
                  box.padding = 0.125, force = 1, force_pull = 1,
                  point.padding = 0); real_fig_3c
ggsave("real_fig_3c.pdf", plot = real_fig_3c, units = "in",
       height = 1.75, width = 1.75) # 1 inch = 72 pt

# fig 3, supp fig 2d: quadrant plot of DAGs colored by ID status
real_fig_3_supp_2d <- ggplot(fig_3b_labels_colors %>%
                   left_join(somata_hetShrink_plot, by = "ensembl_id") %>%
                   filter(label_keep == "yes"), 
       aes(x = log2FoldChange.x, 
           y = log2FoldChange.y,
           color = !(is.na(id_humanID.x)))) +
  geom_vline(xintercept = 0, color = "black", 
             size = 0.25, linetype = "dashed") +
  geom_hline(yintercept = 0, color = "black", 
             size = 0.25,linetype = "dashed") +
  scale_color_manual(values = c("#0072B2", "#D55E00")) +
  theme_classic(base_size = 7, base_family = "sans") + 
  theme(legend.position = c(0.775, 0.7),
        legend.background = element_rect(fill = NA),
        legend.title = element_text(size = 7),
          legend.text = element_text(size = 7),
        legend.spacing.x = unit(0.0002, "in"),
          legend.spacing.y = unit(0.01, "in"),
        axis.text.x = element_text(color = "black"),
          axis.text.y = element_text(color = "black"),
          axis.ticks = element_line(color = "black")) +
  guides(color = guide_legend(title = bquote(ID~Gene), reverse = T,
                              override.aes = list(size = 0.5))) +
  geom_point(size = 0.00000000000000001)  +
  scale_x_continuous(sec.axis = dup_axis(name = "",
                                         breaks = NULL,
                                         labels = NULL)) +
  scale_y_continuous(sec.axis = dup_axis(name = "",
                                         breaks = NULL,
                                         labels = NULL)) +
  labs(x = bquote(GC~log[2](FC)),
       y = bquote(Soma~log[2](FC))) +
    geom_text_repel(data = fig_3b_labels_colors %>%
                   left_join(somata_hetShrink_plot, by = "ensembl_id") %>%
                   filter(!(is.na(id_humanID.x))),
                  aes(label = symbol.x), 
                  size = 1.25, show.legend = F,
                  box.padding = 0.1, force = 1, force_pull = 1,
                  point.padding = 0, nudge_y = 0, nudge_x = -0.5) +
      geom_text_repel(data = fig_3b_labels_colors %>%
                   left_join(somata_hetShrink_plot, by = "ensembl_id") %>%
                   filter(symbol.x == "Ndufa12"),
                  aes(label = symbol.x), 
                  size = 1.25, show.legend = F,
                  box.padding = 0.1, force = 1, force_pull = 1,
                  point.padding = 0, nudge_y = 0.5, nudge_x = 0); real_fig_3_supp_2d
ggsave("real_fig_3_supp_2d.pdf", plot = real_fig_3_supp_2d, units = "in",
       height = 1.75, width = 1.75) # 1 inch = 72 pt

# DAG functional annotations manually generated by omer
omer_function <- read.csv("omer_function.csv")

# fig 3d: DAGs organized by function and colored by log2FC
real_fig_3d <- ggplot(sig_fig_3b_labels_colors %>%
                                 inner_join(omer_function, by = "label"),
                               aes(x = omer_function, y = label,
                                   color = log2FoldChange)) +
  geom_point() +
  scale_color_distiller(palette = "RdYlBu") +
  theme_classic(base_size = 7, base_family = "sans") +
  geom_text(aes(label = symbol),
             color = "black", size = 2.5, nudge_x = 0.25) +
  labs(x = bquote(Function),
       y = bquote(Gene),
       color = bquote(log2FC)); real_fig_3d
ggsave("real_fig_3d.pdf", plot = real_fig_3d, units = "in",
       height = 11, width = 8.5) # 1 inch = 72 pt

# getting additional mouse gene information for venn diagram
mouse_genBank <- as.data.frame(org.Mm.egACCNUM2EG) %>%
  rename(Genbank = accession) %>%
  right_join(mouse_symbols, by = "gene_id")

# subcellular loclization from Zivraj et al., J.Neurosci, 2010
# https://doi.org/10.1523/JNEUROSCI.1800-10.2010
# it's called stupid because i had to import their tables into excel as pdfs :/
stupid <- read.csv("mouse_stupid.csv") %>%
  rename(symbol = Common) %>%
  left_join(mouse_genBank, by = "symbol") %>%
  drop_na(ensembl_id) %>%
  distinct(ensembl_id)

# subcellular localization from Poulopoulous* & Murphy* et al., Nature, 2019
# https://doi.org/10.1038/s41586-018-0847-y
pou <- read.csv("pou.csv") %>%
  filter((RNA.GC.Soma > 0 & 
            RNA.GC.vs.Soma.Significant.5..FDR == "+") |
           (Protein.GC.Soma > 0 & 
              Protein.GC.vs.Soma.significant.5..FDR == "+")) %>%
  separate_rows(ENSG) %>%
  distinct(ENSG)

# subcellular localization for DAGs
bcl11a_dags <- sig_fig_3b_labels_colors %>%
  distinct(ensembl_id)

# fig 3, supp fig 2 e: venn diagram of DAGs compared to other GC molecules
venn.diagram(x = list(stupid$ensembl_id, pou$ENSG, 
                      bcl11a_dags$ensembl_id),
             category.names = c("stupid", "pou", "bcl11a_dags"),
             filename = "real_fig_3_supp_2e.png",
             output = T)
```

## UTR Analysis

```{r UTRs}
# getting utr sequences for all detected transcripts
mouse_utr <- getBM(attributes = c("ensembl_gene_id", "3_utr_start", "3_utr_end",
                                  "start_position", "end_position",
                                  "ensembl_transcript_id"),
                   filters = "ensembl_gene_id",
                   values = fig_3b_labels_colors$ensembl_id,
                   mart = mouse_mart) %>%
  mutate(utr_length_3 = `3_utr_end` - `3_utr_start`,
         gene_length = end_position - start_position)
mouse_ref_ensembl <- getBM(attributes = c("ensembl_transcript_id", "refseq_mrna"),
                   filters = "ensembl_transcript_id",
                   values = mouse_utr$ensembl_transcript_id,
                   mart = mouse_mart)
utr3_sequences <- getSequence(seqType = "3utr", mart = mouse_mart,
                              type = "refseq_mrna",
                              id = mouse_ref_ensembl$refseq_mrna)
trans_utr3 <- utr3_sequences %>%
  filter(`3utr` != "Sequence unavailable") %>%
  left_join(mouse_ref_ensembl, by = "refseq_mrna") %>%
  left_join(mouse_utr, by = "ensembl_transcript_id") %>%
  rename(ensembl_id = ensembl_gene_id,
         refseq = refseq_mrna,
         seq = `3utr`) %>%
  left_join(fig_3b_labels_colors, by = "ensembl_id") %>%
  mutate(seq_type = "3UTR") %>%
  unite(trans_name, refseq, seq_type, remove = F, sep = "|")

# background transcripts for transite.mit.edu
write.table(mouse_ref_ensembl %>%
              select(refseq_mrna) %>%
              filter(refseq_mrna != ""),
            "background_utrs.txt",
            quote = F, row.names = F, sep = "\t")

# foreground transcripts for transite.mit.edu
write.table(trans_utr3 %>%
              filter(significant == T) %>%
              select(refseq, direction), "trans_utr3.txt",
            quote = F, row.names = F, sep = "\t")

# analyzing 3' UTR transite results for depleted and enriched DAGs
trans_utr3_depleted <- read.table("trans_utrs_3_pt5/motifs_Depleted.txt",
                                  sep = "\t", header = T) %>%
  mutate(utr_significant = case_when(adj_p_value < 0.05 ~ T,
                                 TRUE ~ F),
         motif_rbps = strsplit(as.character(motif_rbps), ",")) %>%
  unnest(motif_rbps) %>%
  rename(symbol = motif_rbps) %>%
  left_join(mouse_human, by = "symbol")
trans_utr3_enriched <- read.table("trans_utrs_3_pt5/motifs_Enriched.txt",
                                  sep = "\t", header = T) %>%
  mutate(utr_significant = case_when(adj_p_value < 0.05 ~ T,
                                 TRUE ~ F),
         motif_rbps = strsplit(as.character(motif_rbps), ",")) %>%
  unnest(motif_rbps) %>%
  rename(symbol = motif_rbps) %>%
  left_join(mouse_human, by = "symbol")
```

## THE END

```{r theEnd, eval = F}
# THE END
```

